Modules/_testcapimodule.c

```diff
@@ -870,6 +873,45 @@ test_thread_state(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 }
 
+#ifndef MS_WINDOWS
+static PyThread_type_lock wait_done = NULL;
+
+static void wait_for_lock(void *unused) {
+    PyThread_acquire_lock(wait_done, 1);
+}
+
+// These can be used to test things that care about the existence of another
+// thread that the Python runtime doesn't know about.
```

Comment:
```suggestion
// thread that the threading module doesn't know about.
```

The Python runtime does know about these threads (they hold the GIL even :). It would be kinda nice if we could tell whether any (non-main) PyThreadStates were active, since that would detect these kinds of threads too. I guess we want to be careful not to warn when the extra threads have recently exited and _right now_ there is only one active thread.

Reply:
The thread this pair launches and tells to exit never touches the GIL.

The PyThread APIs are not APIs that use the GIL, they're low level CPython internal only things that merely abstract platform thread and lock calls, which will only ever be pthread_create and friends given my use is POSIX specific. Using them avoided me needing to figure those out and possibly carry more ifdef conditionals in here.

I suppose that means someone could use these in a non-posix test such as on Windows for another purpose if desired. If we ever refactored this into a Python API to return the current number of threads in the process, that'd presumably want to use it.